### ISSUE TITLE
Enforce semver format for `version`

### ISSUE THREAD

**Issue by JBBianchi** [serverlessworkflow/specification#688]
**What would you like to be added**:
The `version` of a [workflow definition](https://github.com/serverlessworkflow/specification/blob/main/specification.md#workflow-definition-structure) can be any string. Enforcing [semver](https://semver.org/) should be considerated.

**Why is this needed**:
In my understanding, the `version` is used to distinguish different iterations of a said workflow. For instance, a user creates a workflow named `Beautiful Workflow`. After some time, something needs to be changed for whatever reason. The user updates the `Beautiful Workflow`. There are two versions of the same workflow living side by side, for instance 1.0.0 and 1.1.0.
Currently, a version could be anything, a user could deploy workflows such as `{ "name": "pokemon", "version": "red" }` and `{ "name": "pokemon", "version": "blue" }`. I'm not sure that it's an intended use of this field. 

WDYT ?

**Comment by tsurdilo** [comment]
Hey, remember we had these discussions before and ended up with not enforcing a particular versioning style. 
Worth discussing again so lets get everyones opinions here.

**Comment by JBBianchi** [comment]
Sorry if I'm bringing back a dead thread, feel free to close the issue. Your take on it @tsurdilo is to have "free" version?

**Comment by tsurdilo** [comment]
Let's keep it open to gather peoples opinions as it might be relevant.
At the time we discussed it some users mentioned they have very customized versioning needs and said enforcing would not fit their use case, thats why I remember we did not enforce (I was fan of semver too by the way :) )

**Comment by tsurdilo** [comment]
Here is a comment I made on an old issue from last year:

"The idea in the text is that if you use a standard such as SemVer rather than some custom versioning scheme, there is a higher probability chance it will be supported by more runtimes"

**Comment by tsurdilo** [comment]
It's also mentioned in spec documentation in this section: https://github.com/serverlessworkflow/specification/blob/main/specification.md#workflow-versioning

**Comment by ricardozanini** [comment]
Agreed to keep an open string. IMHO, there's no strong reason for the spec to enforce such a pattern.

**Comment by ricardozanini** [comment]
Not sure. On Kubernetes, for instance, they use `v1`, `v1alpha1`, and so on. Each user might have a way of setting up versions. I've seen things like `20220922` or even branch naming. I'd let the conventions be for users.

**Comment by cdavernas** [comment]
Bad example: K8s uses semver ;) (like github, gitlab, knative, argo, etc.)And versions you supplied are valid semver suffixes.

**Comment by ricardozanini** [comment]
Bah, so bite me. :D

I never used these versions and have always relied on `v.1.2.3` my whole life (IIRC semver recommend this approach). If semver is that permissive, I don't see why not.

**Comment by ricardozanini** [comment]
@cdavernas not sure if they are compatible, at least from the regexp pov: https://regex101.com/r/vkijKf/1/

Test the examples I had. From a SDK perspective, I would apply this regexp, which will fail if someone out there decide to use one of them.

**Comment by cdavernas** [comment]
@ricardozanini yeah, `alpha` alone is indeed invalid, but it's a valid suffix: `1.0.0-alpha` for example.

And yeah, I feel you man, I likewise used single then four digits versions a lot until I ran into semver.

**Comment by ricardozanini** [comment]
As discussed in today's meeting, +1 from my side will potentially include this in the next release. It will not only help implementations to have a straightforward way to control the workflow instances version as business to follow a clear standard when defining their workflows.

@tsurdilo do you mind sharing your thoughts?

**Comment by tsurdilo** [comment]
I'm all for it. small change and would bring imo good value for portability

**Comment by ricardozanini** [comment]
@JBBianchi, are you opening a PR?

**Comment by JBBianchi** [comment]
@ricardozanini sure, I'll try to open a draft asap.
